ddtrace/tracer/post_processor.go

```diff
@@ -0,0 +1,111 @@
+package tracer
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// Note: Keeping everything in this file for easy access during development.
+// I will likely move the ReadWriteSpan interface to the ddtrace package,
+// and the (s *span) *() methods to span.go.
+
+// ReadOnlySpan specifies methods to read from a span.
+type readOnlySpan interface {
+	// Span name returns the operation of the span.
+	SpanName() string
+	// etc...
+}
+
+// ReadWriteSpan implements the methods of ddtrace.Span (to write to a span)
+// and readOnlyspan (to read from a span).
+type ReadWriteSpan interface {
+	ddtrace.Span
+	readOnlySpan
+	Remove()
```

Comment:
Why do we need `Remove`? Let's let users decide how to remove things from a slice. They have access to the slice. Main reason is that it doesn't make sense in the context of `Span` to have a `Remove` method (e.g. what are we removing from the span?)

Reply:
Sounds good, will let the customer decide how to remove things from the slice. In this case, I will need to change the processor function to: `func([]ReadWriteSpan) []ReadWriteSpan` where the customer returns the new slice, or nil if they want to drop the trace. Does that sound good ?

Reply:
You're right, I missed that detail. Let's keep it then, and call it `Drop()`